src/passes/EnableSharedMemory.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "pass.h"
+#include "wasm-features.h"
+
+namespace wasm {
+
+struct EnableSharedMemory : public Pass {
+  void run(PassRunner* runner, Module* module) override {
+    if (module->memory.shared) {
+      return;
+    }
+
+    module->memory.shared = true;
+    module->features.set(FeatureSet::Atomics, true);
```

Comment:
We generally avoid adding new features that were not present in the original module or added on the command line, so perhaps this should instead check whether `atomics` is enabled and print a warning if it not?

Reply:
Without it validation fails in [wasm-validator.cpp](https://github.com/WebAssembly/binaryen/blob/6247e7cd9be619d53c926975690981aa267917f9/src/wasm/wasm-validator.cpp#L3036-L3040)

```
  if (curr.shared) {
    info.shouldBeTrue(module.features.hasAtomics(),
                      "memory",
                      "memory is shared, but atomics are disabled");
  }
```

I am not sure if it affects something. Can I turn it into warning?

Reply:
I have something like this in mind:

```
if (atomics)
  memory.shared = true;
else
  warn("Cannot make the memory shared without --enable-atomics")
```